File: example/src/params.rs
use nih_plug::prelude::*;
use react_plug::prelude::*;

define_params! {
    ExampleParams {
        gain: FloatParam {
            name: "Gain",
            default_value: util::db_to_gain(0.0),
            range: FloatRange::Skewed {
                min: util::db_to_gain(-60.0),
                max: util::db_to_gain(6.0),
                factor: FloatRange::gain_skew_factor(-60.0, 6.0),
            },
            smoother: SmoothingStyle::Logarithmic(50.0),
            unit: " dB",
            value_to_string: formatters::v2s_f32_gain_to_db(2),
            string_to_value: formatters::s2v_f32_gain_to_db(),
        },
        reversed: FloatParam {
            name: "Gain",
            default_value: util::db_to_gain(0.0),
            range: FloatRange::Reversed (
                &FloatRange::Linear {
                    min: 0.0,
                    max: 1.0
                }
            ),
        },
        bool_test: BoolParam {
            name: "Bool Test",
            default_value: false
        },
        int_test: IntParam {
            name: "Int Test",
            default_value: 0,
            range: IntRange::Linear { min: 0, max: 10 }
        },
        wave_shape: EnumParam {
            name: "Wave Shape",
            default_value: Sine,
            variants: Waveform {
                Sine: "Sine Wave",
                Square: "Square Wave",
                WhiteNoise: "White Noise",
                Dirac
            }
        }
    }
}

File: example/gui/src/bindings/Params.ts
import * as ReactPlug from "@exa04/react-plug";

export type Params = {
    gain: ReactPlug.Parameters.FloatParam, 
    reversed: ReactPlug.Parameters.FloatParam, 
    bool_test: ReactPlug.Parameters.BoolParam, 
    int_test: ReactPlug.Parameters.IntParam, 
    wave_shape: ReactPlug.Parameters.EnumParam
};

export const createParameters: () => Params = () => ({
    gain: new ReactPlug.Parameters.FloatParam({ name: "Gain", defaultValue: 1, range: new ReactPlug.Ranges.SkewedFloatRange({ min: 0.001, max: 1.9952624, factor: 0.1813854 }), unit: " dB", value_to_string: ReactPlug.Formatters.v2s_f32_gain_to_db(2), id: "gain" }), 
    reversed: new ReactPlug.Parameters.FloatParam({ name: "Gain", defaultValue: 1, range: new ReactPlug.Ranges.ReversedFloatRange(new ReactPlug.Ranges.LinearFloatRange({ min: 0, max: 1 })), id: "reversed" }), 
    bool_test: new ReactPlug.Parameters.BoolParam({ name: "Bool Test", defaultValue: false, id: "bool_test" }), 
    int_test: new ReactPlug.Parameters.IntParam({ name: "Int Test", defaultValue: 0, range: new ReactPlug.Ranges.LinearIntRange({ min: 0, max: 10 }), id: "int_test" }), 
    wave_shape: new ReactPlug.Parameters.EnumParam({ name: "Wave Shape", defaultValue: "Sine", variants: { "Sine": "Sine Wave", "Square": "Square Wave", "WhiteNoise": "White Noise", "Dirac": "Dirac" }, id: "wave_shape" })
});
